chainer/links/connection/peephole.py

```diff
@@ -0,0 +1,98 @@
+from chainer.functions.activation import sigmoid
+from chainer.functions.activation import tanh
+from chainer.functions.array import reshape
+from chainer.functions.array import split_axis
+from chainer import link
+from chainer.links.connection import linear
+from chainer import variable
+
+
+class Peephole(link.Chain):
+
+    """Fully-connected LSTM with peephole connections layer.
+
+    This is a fully-connected LSTM layer with peephole connections as a chain.
+    Unlike the :link:`~chainer.links.lstm` link, this chain holds peep_i,
+    peep_f and peep_o as child links besides upward and lateral.
+
+    Given two inputs a previous hidden vector :math:`h` and an input vector
+    :math:`x`, Peephole returns the next hidden vector :math:`h` defined as
+
+    Args:
+        in_size (int): Dimensionality of input vectors.
+        out_size (int): Dimensionality of output vectors.
+
+    Attributes:
+        c (~chainer.Variable): Cell states of LSTM units.
+        h (~chainer.Variable): Output at the previous time step.
+
+    """
+    def __init__(self, in_size, out_size):
+        super(Peephole, self).__init__(
+            upward=linear.Linear(in_size, 4 * out_size),
+            lateral=linear.Linear(out_size, 4 * out_size, nobias=True),
+            peep_i=linear.Linear(out_size, out_size, nobias=True),
```

Comment:
Yes, in this case also, if peephole connections had bias vectors, they would be redundant.
As I described in peephole.py, forward computation is as below.
```
i = sigma(upward(x) + lateral(h) +peep_i(c))
f = sigma(upward(x) + lateral(h) +peep_f(c))
o = sigma(upward(x) + lateral(h) +peep_o(c))
```
As you can see, we do not have to add bias vector.